src/modules/pos/validator/store/actions/staking.js

```diff
@@ -108,10 +108,21 @@ export const stakesRetrieved = () => async (dispatch, getState) => {
   const { network, account } = getState();
   const address = account.current?.metadata?.address;
   try {
-    const votes = await getVotes({ network, params: { address } });
+    const stakes = await getStakes({ network, params: { address } });
+    const validators = await getValidatorList({
+      addresses: stakes.data.map(({ delegateAddress }) => delegateAddress),
+    });
+    const mapValidatorToAddress = validators.data.reduce(
+      (result, validator) => ({ ...result, [validator.address]: validator }),
+      {}
+    );
```

Comment:
looks like this logic can be simplified, the main goal i see here is to get the commission for given address. so just create 

const validatorsCommission = validators.data.reduce(
      (result, validator) => ({ ...result, [validator.address]: validator. commission }),
      {}
    );